### PR TITLE
1671: Messaging: Add HIP version of AsyncOp

### LINKED ISSUE #1671: Support HIP equivalent for AsyncOpCUDA
**What Needs to be Done?**

- [ ] First do this
- [ ] Then do that.
- [ ] Also this other thing.

**Is your feature request related to a problem? Please describe.**

**Describe potential solution outcome**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by JacobDomagala**
Fixes #1671

### CODE REVIEW

File: src/vt/messaging/async_op_hip.h [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+#include "vt/messaging/async_op.h"
+
+namespace vt { namespace messaging {
+
+#if __HIPCC__
+
+/**
+ * \struct AsyncOpHIP
+ *
+ * \brief A asynchronous HIP event that VT can poll completion for.
```
Comment (PhilMiller) [inline src/vt/messaging/async_op_hip.h:56]:
```suggestion
 * \brief An asynchronous HIP event on which VT can poll completion
```

**Comment by nlslatt** [comment]
@JacobDomagala @lifflander Is HIP 3.0 sufficient?

**Comment by PhilMiller** [comment]
With a rebase, this looks good to me. Any reason we shouldn't get this merged?

**Comment by PhilMiller** [comment]
With the changes I added, this passes its test on real AMD hardware.

**Review by PhilMiller — APPROVED** [review]

**Comment by JacobDomagala** [comment]
Rebased with recent develop changes

**Pull request merged**